cmd/repo.go

```diff
@@ -73,6 +75,85 @@ of components or the repository itself.`,
 	return cmd
 }
 
+// the `repo sign` sub command
+func newRepoSignCmd(env *meta.Environment) *cobra.Command {
+	type manifestT struct {
+		// Signatures value
+		Signatures []map[string]string `json:"signatures"`
+		// Signed value
+		Signed interface{} `json:"signed"`
+	}
+
+	cmd := &cobra.Command{
+		Use:   "sign <manifest-file> [key-files]",
+		Short: "Add signatures to a manifest file",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return cmd.Help()
+			}
+
+			fi, err := os.Open(args[0])
+			if err != nil {
+				return err
+			}
+			defer fi.Close()
+
+			m := manifestT{}
+			content, err := ioutil.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+			if err := json.Unmarshal(content, &m); err != nil {
+				return err
+			}
+			payload, err := json.Marshal(m.Signed)
+			if err != nil {
+				return err
+			}
+
+			for _, kf := range args[1:] {
+				f, err := os.Open(kf)
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+
+				ki := v1manifest.KeyInfo{}
+				if err := json.NewDecoder(f).Decode(&ki); err != nil {
+					return err
+				}
+
+				id, err := ki.ID()
+				if err != nil {
+					return err
+				}
+
+				sig, err := ki.Signature(payload)
+				if err != nil {
+					return err
+				}
+
+				m.Signatures = append(m.Signatures, map[string]string{
```

Comment:
I think it's better to check for exist signature fist, if the key already signed this manifest, we can just skip it (or maybe overwrite the signature?)